django/core/management/commands/compilemessages.py

```diff
@@ -61,6 +61,12 @@ def handle(self, **options):
             from django.conf import settings
             basedirs.extend(upath(path) for path in settings.LOCALE_PATHS)
 
+        # Walk entire tree, looking for locale directories
+        for dirpath, dirnames, filenames in os.walk('.', topdown=True):
+            for dirname in dirnames:
+                if dirname == 'locale':
+                    basedirs.insert(0, os.path.join(dirpath, dirname))
```

Comment:
Looks like I just copied this from `makemessages` doing that, but as you say here it immediately becomes a set. I'll change to append :)